File: drm-webui/src/modules/Receipt.ts
import Constant from "~/constant/Constant.ts";

class Receipt {
    id: number;
    status: string;
    transactionHash: string;
    blockHash: string;
    blockNumber: number;
    gasUsed: number;
    deployer: string;
    deployIndex: number;
    createTime: number;

    //all-args
    constructor(
        id: number,
        status: string,
        transactionHash: string,
        blockHash: string,
        blockNumber: number,
        gasUsed: number,
        deployer: string,
        deployIndex: number,
        createTime: number,
    ) {
        this.id = id;
        this.status = status;
        this.transactionHash = transactionHash;
        this.blockHash = blockHash;
        this.blockNumber = blockNumber;
        this.gasUsed = gasUsed;
        this.deployer = deployer;
        this.deployIndex = deployIndex;
        this.createTime = createTime;
    }
}

export const EmptyReceipt = () => {
    return new Receipt(
        Constant.Global.NULL_NUMBER_PLACEHOLDER,
        Constant.Global.NULL_STRING_PLACEHOLDER,
        Constant.Global.NULL_STRING_PLACEHOLDER,
        Constant.Global.NULL_STRING_PLACEHOLDER,
        Constant.Global.NULL_NUMBER_PLACEHOLDER,
        Constant.Global.NULL_NUMBER_PLACEHOLDER,
        Constant.Global.NULL_STRING_PLACEHOLDER,
        Constant.Global.NULL_NUMBER_PLACEHOLDER,
        Constant.Global.NULL_NUMBER_PLACEHOLDER,
    )
}

export default Receipt

File: drm-core/src/main/kotlin/moe/_47saikyo/drm/core/domain/Receipt.kt
package moe._47saikyo.drm.core.domain

import moe._47saikyo.drm.core.constant.GlobalConstant

data class Receipt(
    val id: Long = 0,
    val status: String = GlobalConstant.NULL_PLACEHOLDER,
    val transactionHash: String = GlobalConstant.NULL_PLACEHOLDER,
    val blockHash: String = GlobalConstant.NULL_PLACEHOLDER,
    val blockNumber: Long = 0,
    val gasUsed: Long = 0,
    val deployer: String = GlobalConstant.NULL_PLACEHOLDER,
    val deployIndex: Long = 0,
    val createTime: Long = 0,
)